packages/headless/src/controllers/recommendation/headless-recommendation.ts

```diff
@@ -32,14 +36,63 @@ export interface RecommendationListProps {
 /**
  * The `RecommendationList` controller retrieves information about the current recommendations by the search API, if there are any.
  */
-export type RecommendationList = ReturnType<typeof buildRecommendationList>;
-/** The state relevant to the `RecommendationList` controller.*/
-export type RecommendationListState = RecommendationList['state'];
+export interface RecommendationList extends Controller {
+  /**
+   * Gets new recommendations.
+   */
+  refresh(): void;
 
+  /**
+   * The state relevant to the `RecommendationList` controller.
+   * */
+  state: RecommendationListState;
+}
+
+export interface RecommendationListState {
+  /**
+   * The current error for the last executed query, or `null` if none is present.
+   * */
+  error: RecommendationListError | null;
+
+  /**
+   * `true` if a search is in progress and `false` otherwise.
+   * */
+  isLoading: boolean;
+
+  /**
+   * The recommendations based on the last executed query.
+   * */
+  recommendations: Result[];
+}
+
+export interface RecommendationListError {
```

Comment:
I noticed myself duplicating this interface which is the same as the error exposed by the `QueryError` controller [here](https://github.com/coveo/ui-kit/pull/539/files#diff-14d2f566628ebec2e60b1e7de8ae6e2cbdfe072c4ec10c4e4a11f47c88659db6R30).

I initially went down this path because:
1. The internal interface is called `SearchAPIErrorWithStatusCode` and I do not want to expose the internal api name in the interface since this could change.
2. These error objects may evolve in different ways?

An alternative to reduce the duplication could be to either:
1. Rename the internal interface to something more general (e.g. `ErrorWithStatusCode`), or
2. Create a new controller-level interface with this name and use that for all controllers that expose error objects.

Any thoughts around duplicating vs. not duplicating?

Reply:
So the name `SearchAPIErrorWithStatusCode` is because it is the way the search API returns error, which might not be applicable for all services.

For example, UA is a bit different.

If we want to avoid duplication, I would suggest option #2, where we (try to) map from the internal error object returned by each API to a standard interface.

Reply:
Ok, so I will keep the `SearchAPIErrorWithStatusCode` interface as it is for now, and create new interface with a general name and export that. If we decide to use the general name internally, we will be able to without needing a breaking change.